test/regression/issue/014865.test.ts

```diff
@@ -0,0 +1,8 @@
+import { test, expect } from "bun:test";
+import { Request } from "node-fetch";
+
+test("node fetch Request URL field is set even with a valid URL", () => {
+  expect(new Request("/").url).toBe("/");
```

Comment:
We probably want to check that this only works on node-fetch `Request`, e.g. if you try this on `globalThis.Request`, it throws